Add Arm64 Jobs in Travis-Ci

odidev opened the pull request:
Hi,

Package Owner: Mohneesh Jha
PR change Details:

#Patch Details : .travis.yml, libutils/buffer.c and libutils/string_lib.c have been modified :
.travis.yml : Arm64 jobs has been added with the installation of package require for arm64
libutils/buffer.c: Strcmp() workaround is done for arm64.
libutils/string_lib.c: Strcmp() workaround is done for arm64 and typecasting of char to signed is done as by default arm takes char as unsigned char
#Testing Detail :
The testing is done over travis

#Reviewer Comment : < To be filled by Reviewer >

#PR Description :
Here is my commit message : 
Adding Arm64-Support in travis.
    Workaround is done for libutils/buffer.c and libutils/string_lib.c for strcmp() function as strcmp() has different behaviour in arm64 and amd64. In Arm64, strcmp(char* a,char* b) gives the value of -64 instead of -1.
    In line 65 of libutils/string_lib.c typecasting is done as some architecture consider char as unsigned char and some consider it as signed char such as amd64 consider default char as signed char whereas arm64 consider it as unsigned char.

Body of PR : 
Added ARM64 job in Travis-CI .
- .travis.yml : Arm64 jobs has been added with the installation of package require for arm64
- libutils/buffer.c: Strcmp() workaround is done for arm64.
- libutils/string_lib.c: Strcmp() workaround is done for arm64 and typecasting of char to signed is done as by default arm takes char as unsigned char
 .
Signed-off-by: odidev@puresoftware.com.

#Reviewer: Pruthvi Teja and Shobhit Prasari

Thanks,
Mohneesh Jha

odidev committed:
Add ARM64 job in Travis-CI

- .travis.yml : Added Arm64 jobs along with the installation of package require for arm64

Signed-off-by: odidev <odidev@puresoftware.com>

shobhitpuresoftware commented on libutils/buffer.c:
```diff
@@ -111,7 +111,17 @@ int BufferCompare(const Buffer *buffer1, const Buffer *buffer2)
-            return strcmp(buffer1->buffer, buffer2->buffer);
+            // Adding this as strcmp gives difference of the buffer values at aarch64 whereas it gives 1 or -1 in other platforms, in case compared char buffer were not same.
+            if (buffer1->buffer == buffer2->buffer)
+            {
```
What is the need for this code here? Please justify.
`if (buffer1->buffer == buffer2->buffer) { ... }`
It seems this is useless here.

shobhitpuresoftware commented on libutils/string_lib.c:
```diff
@@ -231,7 +231,14 @@ int StringSafeCompare(const char *const a, const char *const b)
-        return strcmp(a, b);
+        // Adding this as strcmp gives difference of the buffer values at aarch64 whereas it gives 1 or -1 in other platforms, in case compared char buffer were not same.
+        int compare_result = strcmp(a, b);
+        if (compare_result == 0)
```
Keep the same changes as you are doing in the file: libutils/buffer.c

Specially regarding this "if condition"

shobhitpuresoftware commented on libutils/buffer.c:
```diff
@@ -111,7 +111,17 @@ int BufferCompare(const Buffer *buffer1, const Buffer *buffer2)
              * C String comparison
              */
-            return strcmp(buffer1->buffer, buffer2->buffer);
+            // Adding this as strcmp gives difference of the buffer values at aarch64 whereas it gives 1 or -1 in other platforms, in case compared char buffer were not same.
```
Break this comment into two lines ...
or add a 'new-line' before 'whereas'
Need to add '//' too, in new line

odidev committed:
Fixed strcmp's return value issue for ARM64

 - libutils/[buffer.c, string_lib.c]: Fixed strcmp() issue for arm64
   - strcmp() has different behavior in arm64 and amd64.
   - In ARM64, strcmp(char* a, char* b) returns the value other than 1 or -1, as per given char buffer inputs.

Signed-off-by: odidev <odidev@puresoftware.com>

odidev committed:
Upgrading The Distro From Xenial To Bionic

Signed-off-by: odidev <odidev@puresoftware.com>
